### ISSUE TITLE
Add Instance Methods to JAX Frontend

### ISSUE THREAD

**Issue by jkeane508** [unifyai/ivy#3617]
Add [Instance Methods](https://jax.readthedocs.io/en/latest/jax.numpy.html#jax-devicearray) to JAX Frontend

- [ ] T
- [ ] all
- [ ] any
- [ ] argmax
- [ ] argmin
- [ ] argpartition
- [ ] argsort
- [ ] astype
- [ ] broadcast
- [ ] broadcast_in_dim
- [ ] choose
- [ ] clone
- [ ] conj
- [ ] conjugate
- [ ] copy 
- [ ] cumprod
- [ ] cumsum
- [ ] delete
- [ ] diagonal
- [ ] dot
- [ ] flatten
- [ ] imag
- [ ] is_deleted
- [ ] max
- [ ] mean
- [ ] min
- [ ] nonzero
- [ ] platform
- [ ] prod
- [ ] ptp
- [ ] ravel
- [ ] real
- [ ] repeat
- [ ] round
- [ ] searchsorted
- [ ] sort
- [ ] split
- [ ] squeeze
- [ ] std
- [ ] sum
- [ ] swapaxes
- [ ] take
- [ ] trace
- [ ] var

**Comment by devsugun** [comment]
> Add [Instance Methods](https://jax.readthedocs.io/en/latest/jax.numpy.html#jax-devicearray) to JAX Frontend
> 
> * [ ]  T
> * [ ]  all
> * [ ]  any
> * [ ]  argmax
> * [ ]  argmin
> * [ ]  argpartition
> * [ ]  argsort
> * [ ]  astype
> * [ ]  broadcast
> * [ ]  broadcast_in_dim
> * [ ]  choose
> * [ ]  clone
> * [ ]  conj
> * [ ]  conjugate
> * [ ]  copy
> * [ ]  cumprod
> * [ ]  cumsum
> * [ ]  delete
> * [ ]  diagonal
> * [ ]  dot
> * [ ]  flatten
> * [ ]  imag
> * [ ]  is_deleted
> * [ ]  max
> * [ ]  mean
> * [ ]  median
> * [ ]  min
> * [ ]  msort
> * [ ] nanvar
> * [ ]  nonzero
> * [ ]  ones_like
> * [ ]  platform
> * [ ]  prod
> * [ ]  ptp
> * [ ]  ravel
> * [ ]  real
> * [ ]  repeat
> * [ ] reshape
> * [ ] resize
> * [ ]  round
> * [ ] roll
> * [ ]  searchsorted
> * [ ]  sort
> * [ ] sometrue
> * [ ]  split
> * [ ]  squeeze
> * [ ]  std
> * [ ]  sum
> * [ ]  swapaxes
> * [ ]  take
> * [ ]  trace
> * [ ]  var

**Comment by Adityahulk** [comment]
[]#5823

**Comment by FaikYY** [comment]
#7345

**Comment by saif-ashraf99** [comment]
#7440

**Comment by SaifAshraf70** [comment]
#7443

**Comment by xoiga123** [comment]
- [ ] #12082

**Comment by RakshitKumar04** [comment]
Hey, @hirwa-nshuti can we work on these to-dos?

**Comment by shireenchand** [comment]
#13377

**Comment by shireenchand** [comment]
https://github.com/unifyai/ivy/issues/13377

**Comment by rodlaf** [comment]
- [] #13629

**Comment by Sheila-nk** [comment]
- [ ] #13758

**Comment by felixreynoso** [comment]
- [ ] #13887

**Comment by RakshitKumar04** [comment]
- [ ] #14167

**Comment by RakshitKumar04** [comment]
- [ ] #14170

**Comment by Ranaihab** [comment]
diagonal #14257

**Comment by fahad12398** [comment]
- [ ] #14788

**Comment by mRahmanCodes** [comment]
min #14260

**Comment by mRahmanCodes** [comment]
- [ ] #14260

**Comment by VictorOdede** [comment]
- [ ] #14876

**Comment by VictorOdede** [comment]
- [ ] #14936

**Comment by VictorOdede** [comment]
- [ ] #14937

**Comment by RakshitKumar04** [comment]
- [ ] #14953

**Comment by RakshitKumar04** [comment]
- [ ] #14954

**Comment by VictorOdede** [comment]
- [ ] #15368

**Comment by VictorOdede** [comment]
- [ ] #15370

**Comment by VictorOdede** [comment]
- [ ] #15371

**Comment by RakshitKumar04** [comment]
- [ ] #15383

**Comment by RakshitKumar04** [comment]
- [ ] #15387

**Comment by gamatos** [comment]
- [ ] #15715

**Comment by Yarkyka** [comment]
- [ ]  #15716

**Comment by RakshitKumar04** [comment]
Hey, @hirwa-nshuti the following functions have been implemented and merged successfully. 
Can you please close the issues as completed.
- #14170
- #14954 
- #15383 
- #15387

**Comment by VictorOdede** [comment]
- [ ] #16008

**Comment by VictorOdede** [comment]
- [ ] #16009

**Comment by VictorOdede** [comment]
- [ ] #16010

**Comment by VictorOdede** [comment]
- [ ] #16011

**Comment by baberabb** [comment]
- [ ] #16956

**Comment by selvaraj-sembulingam** [comment]
#17008

**Comment by Nanda198** [comment]
-[] #17964

**Comment by rohitkg83** [comment]
- [ ] #18087

**Comment by Rashed6122** [comment]
ref #18187

**Comment by moezAhmad** [comment]
[imag] #18226

**Comment by samthakur587** [comment]
#18242

**Comment by samthakur587** [comment]
#18243

**Comment by MuhammadNizamani** [comment]
- [ ] #18979

**Comment by MuhammadNizamani** [comment]
- [ ]  #19048

**Comment by MuhammadNizamani** [comment]
- [ ] #19210

**Comment by VictorOdede** [comment]
- [ ] #19417

**Comment by VictorOdede** [comment]
- [ ] #19418

**Comment by seemanitharwal** [comment]
- [ ] #19600

**Comment by shakikk** [comment]
Real #19710

**Comment by shakikk** [comment]
real #19710

**Comment by seemanitharwal** [comment]
- [ ] #19542

**Comment by BrolyDBZ** [comment]
#21027

**Comment by yashj8** [comment]
-[] #21279

**Comment by waqaarahmed** [comment]
- [ size] #25555

**Comment by jyjoshi** [comment]
- [ ] #21898

**Comment by candidMaster** [comment]
- [ ] #22278

**Comment by AbdullahSabry** [comment]
- [ ] #22661

**Comment by vincenzofanizza** [comment]
- [ ] #23088

**Comment by gauthamk02** [comment]
- [] #23168

**Comment by gauthamk02** [comment]
-[] #23168

**Comment by gauthamk02** [comment]
- [] #23168

**Comment by gauthamk02** [comment]
- [ ] #23168

**Comment by mcnugets** [comment]
- [ ] #23602

**Comment by Mghrabi** [comment]
#25760

**Comment by ChibuezeOnejeme** [comment]
- [ ]#25877

**Comment by akshatvishu** [comment]
@Ishticode @jkeane508  I don't think `split` is available as a instance method at `jax` and thus, should be removed from this list. 

refs:<https://github.com/google/jax/blob/4471abe1c6874a650a60a975d9e9dd21b010f059/jax/_src/basearray.pyi#L119>

**Comment by AnnaTz** [comment]
> @Ishticode @jkeane508 I don't think `split` is available as an instance method at `jax` and thus, should be removed from this list.
> 
> refs:https://github.com/google/jax/blob/4471abe1c6874a650a60a975d9e9dd21b010f059/jax/_src/basearray.pyi#L119

You are right. I removed it. along with some others that should not exist either.

**Comment by mcnugets** [comment]
> > @Ishticode @jkeane508 I don't think `split` is available as an instance method at `jax` and thus, should be removed from this list.
> > refs:https://github.com/google/jax/blob/4471abe1c6874a650a60a975d9e9dd21b010f059/jax/_src/basearray.pyi#L119
> 
> You are right. I removed it. along with some others that should not exist either. Thanks!



I was attempting to implement the `split` instance method, and couldn't fix reoccurring test errors, but I guess this resolves it. Thank you @AnnaTz

**Comment by mcnugets** [comment]
- [ ] #26512
